cardano-ping/src/Cardano/Network/Ping.hs

```diff
@@ -0,0 +1,445 @@
+        (14, True)  -> Right . NodeToClientVersionV14 <$> CBOR.decodeWord32
+        _           -> return $ Left $ UnknownVersionInRsp version
+
+    decodeWithMode :: (Word32 -> Bool -> NodeVersion) -> CBOR.Decoder s (Either HandshakeFailure NodeVersion)
+    decodeWithMode vnFun = do
+      _ <- CBOR.decodeListLen
+      magic <- CBOR.decodeWord32
+      Right . vnFun magic <$> CBOR.decodeBool
+
+wrap :: MiniProtocolNum -> MiniProtocolDir -> LBS.ByteString -> MuxSDU
+wrap ptclNum ptclDir blob = MuxSDU
+  { msHeader = MuxSDUHeader
+    { mhTimestamp = RemoteClockModel 0
+    , mhNum       = ptclNum
+    , mhDir       = ptclDir
+    , mhLength    = fromIntegral $ LBS.length blob
+    }
+  , msBlob = blob
+  }
+
+
+data StatPoint = StatPoint
+  { spTimestamp :: UTCTime
+  , spHost      :: String
+  , spCookie    :: Word16
+  , spSample    :: Double
+  , spMedian    :: Double
+  , spP90       :: Double
+  , spMean      :: Double
+  , spMin       :: Double
+  , spMax       :: Double
+  , spStd       :: Double
+  }
+
+instance Show StatPoint where
+  show :: StatPoint -> String
+  show StatPoint {..} =
+    printf "%36s, %-28s, %7d, %7.3f, %7.3f, %7.3f, %7.3f, %7.3f, %7.3f, %7.3f"
+      (show spTimestamp) spHost spCookie spSample spMedian spP90 spMean spMin spMax spStd
+
+instance ToJSON StatPoint where
+  toJSON :: StatPoint -> Value
+  toJSON StatPoint {..} =
+    object
+      [ "timestamp" .= spTimestamp
+      , "host"      .= spHost
+      , "cookie"    .= spCookie
+      , "sample"    .= spSample
+      , "median"    .= spMedian
+      , "p90"       .= spP90
+      , "mean"      .= spMean
+      , "min"       .= spMin
+      , "max"       .= spMax
+      ]
+
+toStatPoint :: UTCTime -> String -> Word16 -> Double -> TDigest 5 -> StatPoint
+toStatPoint ts host cookie sample td =
+  StatPoint
+    { spTimestamp = ts
+    , spHost      = host
+    , spCookie    = cookie
+    , spSample    = sample
+    , spMedian    = quantile' 0.5
+    , spP90       = quantile' 0.9
+    , spMean      = mean'
+    , spMin       = minimumValue td
+    , spMax       = maximumValue td
+    , spStd       = stddev'
+    }
+  where
+    quantile' :: Double -> Double
+    quantile' q = fromMaybe 0 (quantile q td)
+
+    mean' :: Double
+    mean' = fromMaybe 0 (mean td)
+
+    stddev' :: Double
+    stddev' = fromMaybe 0 (stddev td)
+
+pingClient :: Tracer IO LogMsg -> Tracer IO String -> PingOpts -> [NodeVersion] -> AddrInfo -> IO ()
+pingClient stdout stderr PingOpts{pingOptsQuiet, pingOptsJson, pingOptsCount} versions peer = bracket
+  (Socket.socket (Socket.addrFamily peer) Socket.Stream Socket.defaultProtocol)
+  Socket.close
+  (\sd -> withTimeoutSerial $ \timeoutfn -> do
+    when (Socket.addrFamily peer /= Socket.AF_UNIX) $ do
+      Socket.setSocketOption sd Socket.NoDelay 1
+      Socket.setSockOpt sd Socket.Linger $ StructLinger
+        { sl_onoff  = 1
+        , sl_linger = 0
+        }
```

Comment:
```suggestion
      Socket.setSockOpt sd Socket.Linger
        StructLinger
          { sl_onoff  = 1
          , sl_linger = 0
          }
```
The `$` is not even needed, keeping the constructor and fields close together is easier to read.

Reply:
Fixed.